WebDriverAgentLib/Commands/FBCustomCommands.m

```diff
@@ -70,4 +72,23 @@ + (NSArray *)routes
   return FBResponseWithOK();
 }
 
++ (id<FBResponsePayload>)handleDismissKeyboardCommand:(FBRouteRequest *)request
+{
+  [request.session.application dismissKeyboard];
+  NSError *error;
+  BOOL isKbdNotPresent =
+  [[[[FBRunLoopSpinner new]
+     timeout:5]
+    timeoutErrorMessage:@"The keyboard cannot be dismissed. Try to use custom approach instead."]
+   spinUntilTrue:^BOOL{
+     XCUIElement *foundKeyboard = [[FBApplication fb_activeApplication].query descendantsMatchingType:XCUIElementTypeKeyboard].element;
+     return !(foundKeyboard.exists && foundKeyboard.hittable);
```

Comment:
`exists` or/and `hittable` will resolve element anyway. I understand 'cannot read attributes' argument, but this is only because is_visible check is broken, which we need to fix anyway. I should have some time next week to work on it, but I'd rather keep code simpler and unified. 

You could add that if to is_visible check, to filter obvious cases :)